File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.java
class Solution {
    public int findMin(int[] nums) {
        int low = 0, high = nums.length - 1;

        while(low < high) {
            int mid = low + (high - low) / 2; 

            if(nums[mid] > nums[high]) {
                // Min is in the right half (exluding mid)
                low = mid + 1; 
            } else {
                // Min is in the left half
                high = mid; 
            }
        }
        return nums[high];
    }
}

File: Data Structures & Algorithms/sort-an-array/submission-2.go
func sortArray(nums []int) []int {
    if len(nums) <= 1 {
		return nums
	}

	mid := len(nums) / 2

	left := sortArray(nums[:mid])
	right := sortArray(nums[mid:])

	return merge(left, right)
}

func merge(left, right []int) []int {
	result := []int{}
	i, j := 0, 0

	for i < len(left) && j < len(right) {
		if left[i] < right[j] {
			result = append(result, left[i])
			i++
		} else {
			result = append(result, right[j])
			j++
		}
	}

	result = append(result, left[i:]...)
	result = append(result, right[j:]...)

	return result
}
